lib/active_model/serializer/fieldset.rb

```diff
@@ -20,11 +20,7 @@ def fields_for(type)
       private
 
       def parsed_fields
-        if raw_fields.is_a?(Hash)
-          raw_fields.each_with_object({}) { |(k, v), h| h[k.to_sym] = v.map(&:to_sym) }
-        else
-          {}
-        end
+        JSONAPI::IncludeDirective.new(raw_fields).to_hash
```

Comment:
alternatively, I could maybe change how the relationship's call to fields_for is handled. maybe it would be a bit more intuitive that way.